common/src/commonMain/kotlin/io/xorum/codeforceswatcher/network/CodeforcesApiClient.kt

```diff
@@ -0,0 +1,94 @@
+package io.xorum.codeforceswatcher.network
+
+import io.ktor.client.HttpClient
+import io.ktor.client.features.defaultRequest
+import io.ktor.client.features.json.Json
+import io.ktor.client.features.json.serializer.KotlinxSerializer
+import io.ktor.client.features.logging.DEFAULT
+import io.ktor.client.features.logging.LogLevel
+import io.ktor.client.features.logging.Logger
+import io.ktor.client.features.logging.Logging
+import io.ktor.client.request.get
+import io.ktor.client.request.parameter
+import io.ktor.client.utils.EmptyContent
+import io.ktor.http.URLProtocol
+import io.xorum.codeforceswatcher.network.responses.*
+import kotlinx.serialization.*
+import kotlinx.serialization.internal.UnitDescriptor
+import kotlinx.serialization.json.Json.Companion.nonstrict
+
+private const val CODEFORCES_API_LINK = "www.codeforces.com/api/"
+
+object CodeforcesApiClient {
+    private val codeforcesApiClient = makeCodeforcesApiClient()
+
+    suspend fun getUsers(handles: String) = try {
+        codeforcesApiClient.get<UsersResponse>(path = "user.info") {
+            parameter("handles", handles)
+        }
+    } catch (t: Throwable) {
```

Comment:
Did you have at least one `Throwable` caught? I'm not sure when it's called when `expectSuccess  = false`.

Reply:
Yeap, I have caught when no internet connection.

Reply:
OK for now.